src/components/implementation/tests/unit_cxx/cxx.h

```diff
@@ -0,0 +1,209 @@
+#ifndef CXX_UNIT_H
+#define CXX_UNIT_H
+
+#include <vector>
+#include <algorithm>
+#include <set>
+#include <queue>
+#include <map>
+#include <string>
+#include <iostream>
+using namespace std;
+#ifdef __cplusplus
+extern "C" {
+#include <print.h>
+#endif
+
+#ifdef __cplusplus
+}
+#endif
+
+class Global_foo
+{
+public:
+	Global_foo();
+	~Global_foo();
+};
+
+Global_foo::Global_foo()
+{
+	printc("Global foo constructor\n");
```

Comment:
Strange. You've `Global_foo` object as a global variable, I wonder if it's c++ lib initialization issue and related to the code here:
https://github.com/gparmer/composite/blob/ppos/src/components/lib/cos_component.c#L61-L65